test/it/code-samples/negative/public-$w-non-string-selector/core.js

```diff
@@ -0,0 +1,8 @@
+const arr = ["firstSelector", "secondSelector"]
+const obj = {
+    test: "testSelector"
+}
+$w(arr).anyProperty();
+$w(obj).z()
+$w(8).testo()
+$w(true).another()
```

Comment:
also, for another PR...
i'm not a big fan of the `public-` prefix that tells the test what context to check.
i would prefer just having a specific folder that  has a structure similar to the real files structure so that i can write a test that verifies how different contexts work together